config/tests/testgrids/config_test.go

```diff
@@ -328,16 +328,15 @@ func TestConfig(t *testing.T) {
 			}
 		}
 
-		// Dashboards that match this dashboard group's prefix should be a part of it
-		for dashboard := range dashboardmap {
-			if strings.HasPrefix(dashboard, dashboardGroup.Name+"-") {
-				group, ok := tabs[dashboard]
-				if !ok {
-					t.Errorf("Dashboard %v should be in dashboard_group %v", dashboard, dashboardGroup.Name)
-				} else if group != dashboardGroup.Name {
-					t.Errorf("Dashboard %v should be in dashboard_group %v instead of dashboard_group %v", dashboard, dashboardGroup.Name, group)
-				}
-			}
+	}
+
+	// All dashboards must be prefixed with their group.
+	for dashboard := range dashboardmap {
```

Comment:
I feel this is a regression from previous logic. Where it tries to match a dashboard with all dashboard group and try to reduce the exploding of the number of dashboard groups.

Reply:
Fixed in subsequent commit.